src/yetibot/api/aws.clj

```diff
@@ -0,0 +1,160 @@
+(ns yetibot.api.aws
+  (:require
+    [yetibot.core.schema :refer [non-empty-str]]
+    [schema.core :as sch]
+    [yetibot.core.config :refer [get-config]]
+    [cognitect.aws.client.api :as aws]
+    [clojure.spec.alpha :as s]))
+
+(def aws-schema
+  {:aws-access-key-id         non-empty-str
+   :aws-secret-access-key     non-empty-str
+   (sch/optional-key :region) non-empty-str})
+
+(defn config
+  "Returns AWS-related configuration"
+  []
+  (:value (get-config aws-schema [:aws])))
+
+(defn configured? [] (config))
+
+; aws API credentials
+(def aws-access-key-id (:aws-access-key-id (config)))
+(def aws-secret-access-key (:aws-secret-access-key (config)))
+
+(defn make-aws-client
+  "Returns an aws client given an aws keywordized service name (:iam, :ec2, :s3 etc.)"
+  [service-name]
+  (aws/client {:api                  service-name
+               :credentials-provider (cognitect.aws.credentials/basic-credentials-provider
+                                       {:access-key-id     aws-access-key-id
+                                        :secret-access-key aws-secret-access-key})}))
+; AWS clients
+(def iam (make-aws-client :iam))
```

Comment:
This could check if config is valid before instantiating to prevent errors.

```clojure
(def iam (when (configured?) (make-aws-client :iam)))
```

Reply:
will refactor that ;-) thanks for the snippet